### PR TITLE
[ENTESB-11495] Mapping variables also on options, not only from the body

### PR DESCRIPTION
**Pull request by Delawen**
Following #6593 now the mapping works fine in all the fields.

### CODE REVIEW

**Comment by Delawen** [comment]
Added a QuickStart with all the functionality: https://github.com/syndesisio/syndesis-quickstarts/pull/37

**Comment by zregvart** [comment]
We have a test failing [on CI](https://app.circleci.com/jobs/github/syndesisio/syndesis/120363):
```
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.763 s <<< FAILURE! - in io.syndesis.connector.aws.ddb.metadata.AWSDDBMetadataAdapterTest

[ERROR] adaptTest(io.syndesis.connector.aws.ddb.metadata.AWSDDBMetadataAdapterTest)  Time elapsed: 0.763 s  <<< FAILURE!

org.junit.ComparisonFailure: expected:<...",\"properties\":{\"[:#KEY]\":{\"type\":\"strin...> but was:<...",\"properties\":{\"[clave]\":{\"type\":\"strin...>

	at io.syndesis.connector.aws.ddb.metadata.AWSDDBMetadataAdapterTest.adaptTest(AWSDDBMetadataAdapterTest.java:78)
```

File: app/connector/aws-ddb/src/test/java/io/syndesis/connector/aws/ddb/metadata/AWSDDBMetadataAdapterTest.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        "    \"kind\" : \"json-schema\",\n" +
+        "    \"type\" : \"Result\",\n" +
+        "    \"specification\" : \"{\\\"type\\\":\\\"object\\\",\\\"title\\\":\\\"Result\\\",\\\"properties\\\":{\\\"clave\\\":{\\\"type\\\":\\\"string\\\"},\\\"atributo\\\":{\\\"type\\\":\\\"string\\\"}}}\",\n" +
+        "    \"metadata\" : {\n" +
+        "      \"variant\" : \"collection\"\n" +
+        "    }\n" +
+        "  }\n" +
+        "}";
+
+        assertEquals(expectedMetadata, actualMetadata);
```
Comment (zregvart) [inline app/connector/aws-ddb/src/test/java/io/syndesis/connector/aws/ddb/metadata/AWSDDBMetadataAdapterTest.java]: Perhaps we could use [JSONAssert](http://jsonassert.skyscreamer.org/) here. Example of the use can be found here:

https://github.com/syndesisio/syndesis/blob/4b628bcbab6fef93684e741a6fae879c729ae31a/app/common/util/src/test/java/io/syndesis/common/util/openapi/OpenApiHelperTest.java#L50-L59

**Comment by Delawen** [comment]
> We have a test failing [on CI](https://app.circleci.com/jobs/github/syndesisio/syndesis/120363):
> 
> ```
> [ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.763 s <<< FAILURE! - in io.syndesis.connector.aws.ddb.metadata.AWSDDBMetadataAdapterTest
> 
> [ERROR] adaptTest(io.syndesis.connector.aws.ddb.metadata.AWSDDBMetadataAdapterTest)  Time elapsed: 0.763 s  <<< FAILURE!
> 
> org.junit.ComparisonFailure: expected:<...",\"properties\":{\"[:#KEY]\":{\"type\":\"strin...> but was:<...",\"properties\":{\"[clave]\":{\"type\":\"strin...>
> 
> 	at io.syndesis.connector.aws.ddb.metadata.AWSDDBMetadataAdapterTest.adaptTest(AWSDDBMetadataAdapterTest.java:78)
> ```

Working on the tests. Classic "on my local it worked" case :) due to non-staged changes.

File: app/connector/aws-ddb/src/test/java/io/syndesis/connector/aws/ddb/metadata/AWSDDBMetadataAdapterTest.java (hunk 1 above)
Comment (Delawen) [reply app/connector/aws-ddb/src/test/java/io/syndesis/connector/aws/ddb/metadata/AWSDDBMetadataAdapterTest.java]: :+1:

**Pull request merged**